proposals/adrianhopebailie-201908.md

```diff
@@ -0,0 +1,639 @@
+transfers.
+
+#### Add `ParticipantList`, `Participant` and `Rate` data model
+
+`ParticipantList`:
+
+| Data Element | Cardinality | Type          | Description                    |
+| ------------ | ----------- | ------------- | ------------------------------ |
+| participant  | 1..16       | `Participant` | Number of Participant elements |
+
+`Participant`:
+
+| Name             | Cardinality | Type               | Description                                                                                                 |
+| ---------------- | ----------- | ------------------ | ----------------------------------------------------------------------------------------------------------- |
+| fspId            | 1           | `FspId`            | The identity of the participant                                                                             |
+| transferCurrency | 1           | `Currency`         | The currency of the transfer that will be made by this participant.                                         |
+| fee              | 1           | `Money`            | The fee that will be charged by the participant.                                                            |
+| rate             | 0..1        | `Rate`             | The rate of exchange that will applied by this participant.                                                 |
+| expiration       | 1           | `DateTime`         | Date and time until when the quotation is valid and can be honored when used in the subsequent transaction. |
+| dataRequired     | 0..1        | `RequiredDataList` | List of data required for compliance                                                                        |
+| encryptionKey    | 0..1        | `JsonWebKey`       | The Public Key of the participant used to encrypt data sent to the participant.                             |
+| dataProvided     | 0..32       | `ProvidedDataList` | The data provided by other participants for use by this participant.                                        |
+
+`Rate`:
+
+| Data Element | Cardinality | Type        | Description                                      |
+| ------------ | ----------- | ----------- | ------------------------------------------------ |
+| rate         | 1           | `RateValue` | Rate of exchange                                 |
+| fromCurrency | 1           | `Currency`  | Currency from which the conversion is being made |
+| toCurrency   | 1           | `Currency`  | Currency to which the conversion is being madre  |
+
+#### `RateValue` data type:
+
+The API data type `RateValue` is a JSON String consisting of digits only and an
+optional period. Negative numbers and leading zeroes are not allowed.
+
+##### `RateValue` Regular Expression
+
+The regular expression for restricting an Rate is
+`^(([1-9][0-9]{0,5}))([.][0-9]{0,8}[1-9])?$`
+
+#### Add `RequiredDataList`, `ProvidedDataList` and `ProvidedData`
+
+The `RequiredDataList` data model is defined as:
```

Comment:
In the case of multi hops, don't we need a way to keep track of _which participant_ is requesting the required data? Or is the plan to keep adding `RequiredData` to the list as the transaction keeps accumulating hops during the Quote request "leg" of the transfer?

Reply:
The list exists for each participant. I.e. Each participant adds to the list of participants which an entry for itself

Reply:
Cool. Thanks for clarifying this for me.